packages/story-editor/src/components/library/panes/text/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,251 @@
+    ({ state, actions }) => {
+      const isText =
+        state.selectedElements && areAllType(TYPE, state.selectedElements);
+      return {
+        isText,
+        textStyles: state.story?.globalStoryStyles?.textStyles || [],
+        updateSelectedElements: actions.updateSelectedElements,
+      };
+    }
+  );
+
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const buttonRef = useRef(null);
+  const stylesRef = useRef(null);
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+  const { isRTL, styleConstants: { topOffset } = {} } = useConfig();
+  const hasPresets = textStyles.length > 0;
+
+  const pushUpdate = useCallback(
+    (update) => {
+      updateSelectedElements({
+        properties: (element) => {
+          const updates = updateProperties(element, update, true);
+          const sizeUpdates = getUpdatedSizeAndPosition({
+            ...element,
+            ...updates,
+          });
+          return {
+            ...updates,
+            ...sizeUpdates,
+          };
+        },
+      });
+    },
+    [updateSelectedElements]
+  );
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
+
+  if (!isStylePaneEnabled) {
+    return null;
+  }
+
+  const addStyledText = (preset) => {
+    insertElement(TYPE, {
+      ...DEFAULT_PRESET,
+      ...preset,
+    });
+  };
+
+  const handlePresetClick = (preset) => {
+    if (isText) {
+      handleApplyStyle(preset);
+    } else {
+      addStyledText(preset);
+    }
+  };
+
+  // If it's not text, we're displaying insertion overlay and label for adding a new text.
+  const styleItemProps = isText
+    ? {}
+    : {
+        activeItemOverlay: <InsertionOverlay />,
+        applyLabel: __('Add new text', 'web-stories'),
+      };
+
+  return (
+    <SectionContainer>
+      <TitleBar>
+        <Headline
+          as="h2"
+          size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.XXX_SMALL}
+        >
+          {__('Saved Styles', 'web-stories')}
+        </Headline>
+        <StyledButton
+          type={BUTTON_TYPES.TERTIARY}
+          size={BUTTON_SIZES.SMALL}
+          variant={BUTTON_VARIANTS.SQUARE}
+          onClick={addGlobalPreset}
+          aria-label={__('Add style', 'web-stories')}
+          disabled={!isText}
+        >
+          <Icons.Plus />
+        </StyledButton>
+      </TitleBar>
+      {hasPresets && (
+        <>
+          <StylesWrapper ref={stylesRef}>
+            {/* We only show the insertion overlay if we're inserting -> this means selected elements are not texts */}
+            <StyleGroup
+              styles={[...textStyles].reverse().slice(0, 2)}
+              handleClick={handlePresetClick}
+              buttonWidth={STYLE_BUTTON_WIDTH}
+              {...styleItemProps}
+            />
+          </StylesWrapper>
```

Comment:
Should these styles be draggable, as the text presets and text sets are? Is there already a new ticket for that? I feel it should be for consistency, but can also see how that could make things more complex because of the reuse of this section in the design panel.